wire-runtime/src/main/java/com/squareup/wire/Message.java

```diff
@@ -160,46 +161,40 @@ protected void setBuilder(Builder builder) {
     return adapter.fromInt(value);
   }
 
+  /**
+   * Create a {@code byte[]} of the serialized form of the data.
+   * <p>
+   * Note: This is an inefficient means of serializing data. For best performance, call
+   * {@link Wire#adapter} and then {@link MessageAdapter2#write}.
+   */
   @SuppressWarnings("unchecked")
   public byte[] toByteArray() {
```

Comment:
I like where this is going. Drop these methods, add methods on MessageAdapter that make it foolishly easy to go from a protocol message to a byte[].